### ISSUE TITLE
Geological Map is rotated and Surfaces are not showing in 3D plot

### ISSUE THREAD

**Issue by AlexanderJuestel** [cgre-aachen/gempy#446]
**Describe the bug**
Using the new plottig API, the geological map does not fit the input data (see image). The issue was observed in two independent projects. 

The geological map was created using:

```ruby
gp.plot_2d(geo_model, section_names=['topography'], show_contours = False)
```

In addition, surfaces do not appear in interactive 3D plots. However, the geological map is oriented correctly in this case. 

The 3D plot was created using: 

```ruby
gpv = gp.plot_3d(geo_model, 
                 show_surfaces=True,
                 show_lith=True, 
                 image=False, 
                 show_results=True,
                 plotter_type='basic', 
                 ve=3)
```

**Expected behavior**
The geological map should fit to the input data and surfaces should be displayed in the 3D plot.

**Screenshots**
Geological map oriented incorrectly
![image](https://user-images.githubusercontent.com/45469915/84744180-3156cb80-afb3-11ea-9cf0-e07bcb7f9c36.png)

Missing surfaces in 3D plot
![image](https://user-images.githubusercontent.com/45469915/84744375-75e26700-afb3-11ea-8950-af2e28a9fe2a.png)

Correctly oriented geological map
![image](https://user-images.githubusercontent.com/45469915/84744402-7ed33880-afb3-11ea-9ffd-03512943ab66.png)

**Desktop (please complete the following information):**
  * OS: Windows 10
  * GemPy Version: 2.2.2

**Comment by AlexanderJuestel** [comment]
@Leguark see this issue for the problem with the rotated geological map ;)

**Comment by wh19971014** [comment]
Hi @AlexanderJuestel ,how the upper model construct,do you have the code making the upper 3D geological model,thank you.

**Comment by AlexanderJuestel** [comment]
Hey @wh19971014,

This was just a sample model. The code originated from the documentation and tutorials of GemPy. I just changed the input data. So, just have a look there. 

Cheers Alex

**Comment by AlexanderJuestel** [comment]
@Leguark, got the same problem for another model now ;) 

![image](https://user-images.githubusercontent.com/45469915/88459099-ef168900-ce92-11ea-860e-d812719175d8.png)

**Comment by endarthur** [comment]
It seems that's an issue with the DEM loading process. If you change the line

https://github.com/cgre-aachen/gempy/blob/233944433890d482265537610618574bd930df33/gempy/plot/visualization_2d.py#L615

to something like

`shape = self.model._grid.topography.values_2d[:, :, 2].shape`

it seems to work (with also transposing the block data after reshape, to correct for the flip).
As far as I have understood, the issue seems to be related with the fact that the grid for the
topography is first created from the block model grid, taking from it its resolution, and that's
not updated after the DEM is loaded using GDAL:

https://github.com/cgre-aachen/gempy/blob/233944433890d482265537610618574bd930df33/gempy/core/grid_modules/topography.py#L90-L93

As set_value doesn't change the resolution attribute.

**Comment by Leguark** [comment]
can some of you to create a test (somewhere in test/test_core/test_grids) replicating one of the models that fail and make a pull request?

I agree with @endarthur that the problem seems to be in the LoadDEMGDAL class but without debugging I cannot tell

**Comment by AlexanderJuestel** [comment]
@endarthur this we need to tackle!

**Comment by AlexanderJuestel** [comment]
@Leguark Do you think we could fix that for the next release that you indicated? It has bothered me a long time now :D But we haven't made any tests yet. 

Maybe @endarthur has the time to create some soon :)

**Comment by AlexanderJuestel** [comment]
Will be fixed in #640
